src/main/java/yapper/app/Task.java

```diff
@@ -0,0 +1,126 @@
+package yapper.app;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import yapper.exceptions.YapperException;
+
+/**
+ * Represents a task with a description and completion status.
+ */
+public class Task {
+    private static String[] timeFormats = {
+        "HHmm",
+        "HH:mm[:ss]"
+    };
+
+    @SuppressWarnings("checkstyle:DeclarationOrder")
+    private static String[] dateFormats = {
+        "d/M/yyyy", // Date only
+        "d/MM/yyyy",
+        "dd/M/yyyy",
+        "dd/MM/yyyy",
+        "yyyy/M/d",
+        "yyyy/MM/d",
+        "yyyy/M/dd",
+        "yyyy/MM/dd",
+    };
+
+    private String desc;
+    private String isDone;
+
+    /**
+     * Constructs a new Task with the specified description.
+     *
+     * @param desc the description of the task
+     * @throws YapperException if the description is empty
+     */
+    public Task(String desc) {
+        if (desc.isEmpty()) {
+            throw new YapperException("Description cannot be empty");
+        }
+        this.desc = desc;
+        this.isDone = "[ ]";
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void mark() {
+        this.isDone = "[X]";
+    }
+
+    /**
+     * Unmarks the task as not completed.
+     */
+    public void unmark() {
+        this.isDone = "[ ]";
+    }
+
+    /**
+     * Formats a date and/or time string into a human-readable format.
+     *
+     * The method first tries to parse the string as a date and time, then as a time only, and finally as a date only.
+     * It returns the formatted date/time or the original string if it cannot be parsed.
+     *
+     * @param time the date and/or time string to be formatted
+     * @return the formatted date/time string, or the original string if parsing fails
+     */
+    public String formattedDate(String time) {
+        for (String timeFormat : timeFormats) {
+            for (String dateFormat : dateFormats) {
+                try {
+                    StringBuilder fm = new StringBuilder(dateFormat).append(" ").append(timeFormat);
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern(fm.toString());
+                    LocalDateTime formattedDateTime = LocalDateTime.parse(time, formatter);
+                    return formattedDateTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a"));
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+            }
+        }
+
+        for (String fm : timeFormats) {
```

Comment:
Maybe format is more preferrable than fm?

Reply:
Will change, thanks.